coalib/settings/ConfigurationGathering.py

```diff
@@ -192,9 +195,14 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
         sections = merge_section_dicts(sections, cli_sections)
 
-    for section in sections:
-        if section != 'default':
-            sections[section].defaults = sections['default']
+    for name, section in list(sections.items()):
+        section.set_default_section(sections)
+        if name == 'default' and not all(
+                setting.from_cli for setting in section.contents.values()):
+            logging.warning('Implicit \'Default\' section inheritance is '
+                            'deprecated. It will be removed soon. To silence '
```

Comment:
I'd add something like `You can use dots to specify inheritance: the section all.python will inherit all settings from all.` so people get to know the new thing right ahead for migration!